### PR TITLE
various improvements

### PR DESCRIPTION
**Pull request by valoq**
Add profile for pinentry-gtk and add various minor fixed for missing permissions as logged by aa-log

The permissions for` /dev/snd/controlC0 r, `and  ` /dev/snd/controlC1 r,` are required by firefox but I added them to audio-client since they may be required by other clients as well

### CODE REVIEW

File: apparmor.d/abstractions/audio-client [hunk 1]
```diff
@@ -60,6 +61,9 @@
         /dev/shm/ r,
   owner /dev/shm/pulse-shm-@{int} rw,
 
+  /dev/snd/controlC0 r,
+  /dev/snd/controlC1 r,
```
Comment (roddhjav) [inline apparmor.d/abstractions/audio-client:65]: Use `@{int}` instead of 0 or 1.

File: apparmor.d/profiles-g-l/libreoffice [hunk 2]
```diff
@@ -71,6 +71,7 @@ profile libreoffice @{exec_path} {
   /etc/libreoffice/{,**} r,
   /etc/paperspecs r,
   /etc/xdg/* r,
+  /etc/machine-id r,
```
Comment (roddhjav) [inline apparmor.d/profiles-g-l/libreoffice:74]: bus-session already includes this.

File: apparmor.d/profiles-g-l/git [hunk 3]
```diff
@@ -18,6 +18,8 @@ profile git @{exec_path} flags=(attach_disconnected) {
   include <abstractions/nameservice-strict>
   include <abstractions/ssl_certs>
 
+  capability dac_read_search,
+
```
Comment (roddhjav) [inline apparmor.d/profiles-g-l/git:22]: This mostly depend of the location of the git repo. For normal location, it should not be needed.

File: apparmor.d/profiles-m-r/pinentry-gtk [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  include <abstractions/base>
+  include <abstractions/consoles>
+  include <abstractions/gtk>
+  include <abstractions/fonts>
+  include <abstractions/fontconfig-cache-read>
+  include <abstractions/freedesktop.org>
+
+  @{exec_path} mr,
+
+  /usr/share/** r,
```
Comment (roddhjav) [inline apparmor.d/profiles-m-r/pinentry-gtk:20]: This is too wide. Restrict it.

File: apparmor.d/profiles-m-r/pinentry-gtk [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+include <tunables/global>
+
+@{exec_path} = @{bin}/pinentry-gtk
+profile pinentry-gtk @{exec_path} {
+  include <abstractions/base>
+  include <abstractions/consoles>
+  include <abstractions/gtk>
+  include <abstractions/fonts>
+  include <abstractions/fontconfig-cache-read>
+  include <abstractions/freedesktop.org>
```
Comment (roddhjav) [inline apparmor.d/profiles-m-r/pinentry-gtk:16]: You probably wand to use `desktop` instead. (keep fontconfig)

**Comment by roddhjav** [comment]
Instead of adding a profile for `pinentry-gtk`, rename `pinentry-gtk-2` to `pinentry-gtk` with support for both path.

File: apparmor.d/profiles-s-z/signal-desktop-chrome-sandbox [hunk 6]
```diff
@@ -27,6 +28,9 @@ profile signal-desktop-chrome-sandbox @{exec_path} {
   @{PROC}/@{pid}/oom_adj w,
   @{PROC}/@{pid}/oom_score_adj w,
 
+  # Silencer
+  deny /dev/pts/0 rw, # file_inherit
```
Comment (roddhjav) [inline apparmor.d/profiles-s-z/signal-desktop-chrome-sandbox:32]: Use `@{int}` instead of 0.

**Comment by valoq** [comment]
Should be resolved now, thanks

**Comment by roddhjav** [comment]
Thanks, merged!

**Pull request merged**